### PR TITLE
Fix Bug 1246806, ensure video element interactions work, remove underline from button hover state

### PR DESCRIPTION
**Pull request by schalkneethling**
@jpetto r?

### CODE REVIEW

File: media/js/mozorg/contribute/taskview.js [hunk 1]
```diff
@@ -203,40 +203,72 @@
...
-                        // once the step has been completed,
-                        // remove the event listener.
-                        $jocVideo.off('timeupdate.taskview');
-                        $jocVideo.data('watched', true);
-                        trackInteraction('completed joy of coding', $this.data('variation'));
-                    }
-                });
-            }
+        // ensure the video is paused before playing and tracking the interaction.
+        if ($this.data('action') === 'play' && videoElement.paused) {
```
Comment (jpetto) [inline media/js/mozorg/contribute/taskview.js]: Is it possible for this `if` block and the `if` block above to both execute? If not, I'd suggest wrapping this `if/else` block in an `else` for clarity.

File: media/js/mozorg/contribute/taskview.js [hunk 2]
```diff
@@ -203,40 +203,72 @@
...
+            videoElement.play();
+
+            $this.attr('data-ga', true);
+            $jocVideo.attr('data-ga', true);
+
+            trackInteraction('play joy of coding', $this.data('variation'));
+            markAsWatched($this);
```
Comment (jpetto) [inline media/js/mozorg/contribute/taskview.js]: I think this should be `markAsWatched($jocVideo)`?

File: media/js/mozorg/contribute/taskview.js (hunk 1 above)
Comment (schalkneethling) [inline media/js/mozorg/contribute/taskview.js]: Yeah, I was in two minds about that, probably safer to if/else

File: media/js/mozorg/contribute/taskview.js (hunk 2 above)
Comment (schalkneethling) [inline media/js/mozorg/contribute/taskview.js]: Oooooh, indeed.

**Comment by jpetto** [comment]
All appears to be working properly now. :+1:

Care to squash?

**Comment by jpetto** [comment]
Looks good! r+

**Pull request merged**